docs/plugins/discovery-ec2.asciidoc

```diff
@@ -1,167 +1,214 @@
-    cluster discovery. (NOTE: You could provide either group NAME or group
-    ID.)
+    A list of the names or IDs of the security groups to use for discovery. The
+    `discovery.ec2.any_group` setting determines the behaviour of this setting.
+    Defaults to an empty list, meaning that security group membership is
+    ignored by EC2 discovery.
 
-`host_type`::
+`discovery.ec2.any_group`::
+
+    Defaults to `true`, meaning that instances belonging to _any_ of the
+    security groups specified in `discovery.ec2.groups` will be used for
+    discovery. If set to `false`, only instances that belong to _all_ of the
+    security groups specified in `discovery.ec2.groups` will be used for
+    discovery.
+
+`discovery.ec2.host_type`::
 
 +
 --
-The type of host type to use to communicate with other instances. Can be
-one of `private_ip`, `public_ip`, `private_dns`, `public_dns` or `tag:TAGNAME` where
-`TAGNAME` refers to a name of a tag configured for all EC2 instances. Instances which don't
-have this tag set will be ignored by the discovery process.
 
-For example if you defined a tag `my-elasticsearch-host` in ec2 and set it to `myhostname1.mydomain.com`, then
-setting `host_type: tag:my-elasticsearch-host` will tell Discovery Ec2 plugin to read the host name from the
-`my-elasticsearch-host` tag. In this case, it will be resolved to `myhostname1.mydomain.com`.
-http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html[Read more about EC2 Tags].
+Each EC2 instance has a number of different addresses that might be suitable
+for discovery. This setting allows you to select which of these addresses is
+used by the discovery process. It can be set to one of `private_ip`,
+`public_ip`, `private_dns`, `public_dns` or `tag:TAGNAME` where `TAGNAME`
+refers to a name of a tag. This setting defaults to `private_ip`.
 
-Defaults to `private_ip`.
---
+If you set `discovery.ec2.host_type` to a value of the form `tag:TAGNAME` then
+the value of the tag `TAGNAME` attached to each instance will be used as that
+instance's address for discovery. Instances which do not have this tag set will
+be ignored by the discovery process.
 
-`availability_zones`::
+For example if you tag some EC2 instances with a tag called
+`my-elasticsearch-host` and set `host_type: tag:my-elasticsearch-host` then the
+`discovery-ec2` plugin will read each instance's host name from the
+`my-elasticsearch-host` tag.
+http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html[Read more
+about EC2 Tags].
 
-    Either a comma separated list or array based list of availability zones.
-    Only instances within the provided availability zones will be used in the
-    cluster discovery.
+--
 
-`any_group`::
+`discovery.ec2.availability_zones`::
 
-    If set to `false`, will require all security groups to be present for the
-    instance to be used for the discovery. Defaults to `true`.
+    A list of the names of the availability zones to use for discovery. Only
```

Comment:
I think "names" is not the right term here? Googling aws availability zones leads to: 
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html , which talks of region code (plus letter) and AZ ID. I suggest to remove "names of " and add this link to documentation.

Reply:
It is strange that they don't give a name for the region-code-plus-letter identifier in those docs, but this is indeed called as the name of the zone elsewhere. For instance, the `describe-availability-zones` API returns JSON of the following form.

```
        ...
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-west-2",
            "ZoneName": "us-west-2a",
            "ZoneId": "usw2-az2"
        },
        ...
```

I added clarifying text and a link in 5e5b35f15e3.